unic/char/range/src/step.rs

```diff
@@ -0,0 +1,40 @@
+use std::char;
+
+const BEFORE_SURROGATE: char = '\u{D7FF}';
+const AFTER_SURROGATE: char = '\u{E000}';
+
+/// Step a `char` toward `char::MAX`.
+///
+/// If the given `char` is already `char::MAX`, it is returned unchanged.
+#[allow(unsafe_code)]
+pub fn forward(c: char) -> char {
+    if c == char::MAX {
```

Comment:
Cool! Yeah, as long as the behavior is intended and works similar to std lib, it's fine.

I think it worth adding a hint about it in the constructors, though.